### DISCUSSION TITLE
Unable to download traces from Azure Devops pipeline

### DISCUSSION THREAD

**Discussion by dreams-oneindia** [allure-framework/allure2#2263] [Questions & Support] [answered]
1. I have my playwright automation project available in Azure DevOps
2. I have added the "Publish Allure Report" task to the pipeline.
3. The report is being displayed in a separate tab in the pipeline, post-execution
4. Traces are available for the failed test cases.
5. If I try to download the traces, it simply opens empty tabs. But **no download happens**

**Comment by baev** [comment]
Please specify the Azure plugin you're using

**Reply by dreams-oneindia** [reply]
<img width="423" alt="image" src="https://github.com/allure-framework/allure2/assets/119049958/d914309d-7136-48ae-91fc-3ebe3dd0ab57">

**Reply by dreams-oneindia** [reply]
If I open it in Windows:
1. Office Chrome account -> opens 3 tabs with URL text "about:blank"
2. Personal Chrome account -> opens 1 tab with URL text "about:blank"

If I open it in Mac:
1. Opens 1 tab with URL text "https://michaelclay.gallerycdn.vsassets.io/extensions/michaelclay/azure-pipelines-allure-report-viewer/1.0.21/1677573881498/dist/tabContent.html"

However, neither anything is visible on any of the pages, nor the traces get downloaded.

**Comment by dreams-oneindia** [comment]
Hi @baev ,

Did you happen to analyze this? I am stuck with this issue. I need the traces in the pipeline.

**Reply by baev** [reply]
The plugin you're using is an unofficial one created by the community. The root of all the problems with Azure Pipelines is that it only supports one file per report, so all the data should be stored in index.html (with blobs stored as base64 strings). I guess the plugin you're using has some issues with packing attachments to the report. 

There is a bug in one of Azure's components related to publishing reports: https://github.com/microsoft/azure-pipelines-task-lib/issues/909 . We created it a while ago, and it's still open.

We do have plans to revisit the Azure integration early next year, to check if there are any other workarounds available for the issue

**Reply by dreams-oneindia** [reply]
Okay, let me try the index.html. Thank You for your prompt response :)

**Reply by Karzone** [reply]
Hi @baev , did you get any luck with this issue, I got stuck with the same issue as well :(

**Comment by baev** [comment, marked as answer]
Official Allure extension for Azure Pipelines are now available https://marketplace.visualstudio.com/items?itemName=qameta.allure-azure-pipelines

History isn't yet supported, but we'll get there! Please submit feedback to our [discussions](https://github.com/orgs/allure-framework/discussions)
